google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -113,12 +113,18 @@ public CancellableRunnable createDeployRunner(
       AppEngineDeploymentConfiguration deploymentConfiguration,
       DeploymentOperationCallback callback) {
 
-    if (source.getFile() == null
-        || !(source instanceof AppEngineDeployable)) {
-      callback.errorOccurred(GctBundle.message("appengine.deployment.source.not.found.error"));
+    if (!(source instanceof AppEngineDeployable)) {
+      callback.errorOccurred(GctBundle.message("appengine.deployment.invalid.source.error"));
       throw new RuntimeException("Invalid deployment source selected for deployment");
     }
 
+    if (source.getFile() == null
+        || !source.getFile().exists()) {
+      callback.errorOccurred(GctBundle.message("appengine.deployment.source.not.found.error",
+          source.getFilePath()));
+      throw new RuntimeException("Deployment source selected for deployment not found");
```

Comment:
I think it makes sense to not throw an exception here and just return. But for the other case (not an instance of AppEngineDeployable) to keep the exception.